tls/s2n_security_policies.c

```diff
@@ -694,3 +688,24 @@ int s2n_connection_is_valid_for_cipher_preferences(struct s2n_connection *conn,
 
     return 0;
 }
+
+int s2n_validate_kem_preferences(const struct s2n_kem_preferences *kem_preferences, bool pq_kem_extension_required) {
+    notnull_check(kem_preferences);
+
+    /* Basic sanity checks to assert that the count is 0 if and only if the associated list is NULL */
+    S2N_ERROR_IF(!S2N_IFF(kem_preferences->kem_group_count == 0, kem_preferences->kem_groups == NULL),
```

Comment:
S2N_ERROR_IF is eventually being removed and replaced with [ENSURE_POSIX](https://github.com/awslabs/s2n/blob/master/utils/s2n_safety.h#L94) for better readability. 

`ENSURE_POSIX(S2N_IFF(kem_preferences->kem_group_count == 0, kem_preferences->kem_groups == NULL),
S2N_ERR_INVALID_SECURITY_POLICY);`

Reply:
Fixed